<?xml version="1.0"?>
<robot name="ur5">
  <link name="base_link">
    <inertial>
      <origin xyz="-0.00036702 -0.00036758 0.016561" rpy="0 0 0"/>
      <mass value="4.0"/>
      <inertia ixx="0.00060258" ixy="1.7502E-05" ixz="1.5277E-06" iyy="0.00060261" iyz="1.5241E-06" izz="0.0010709"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://ur5/base_link.stl"/>
      </geometry>
      <material name="Black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://ur5/base_link.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="shoulder_link">
    <inertial>
      <origin xyz="-0.078966 0.016439 -6.3565E-06" rpy="0 0 0"/>
      <mass value="3.7"/>
      <inertia ixx="0.0028298" ixy="0.00031932" ixz="-1.3132E-07" iyy="0.002538" iyz="4.4907E-07" izz="0.0026207"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://ur5/shoulder_link.stl"/>
      </geometry>
      <material name="Blue">
        <color rgba="0 0 1 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://ur5/shoulder_link.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="shoulder_pan_joint" type="revolute">
    <origin xyz="0 0 0.0099916" rpy="1.5708 1.5708 0"/>
    <parent link="base_link"/>
    <child link="shoulder_link"/>
    <axis xyz="1 0 0"/>
    <limit lower="-6.2832" upper="6.2832" effort="300" velocity="3"/>
    <dynamics damping="10" friction="1"/>
  </joint>
  <link name="upper_arm_link">
    <inertial>
      <origin xyz="-0.21652 0.079293 8.661E-08" rpy="0 0 0"/>
      <mass value="8.393"/>
      <inertia ixx="0.0063671" ixy="-9.2154E-12" ixz="9.0018E-07" iyy="0.021824" iyz="-2.2572E-07" izz="0.021327"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://ur5/upper_arm_link.stl"/>
      </geometry>
      <material name="White">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://ur5/upper_arm_link.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="shoulder_lift_joint" type="revolute">
    <origin xyz="-0.080008 0.053821 0" rpy="0 0 0"/>
    <parent link="shoulder_link"/>
    <child link="upper_arm_link"/>
    <axis xyz="0 -1 0"/>
    <limit lower="-1.57" upper="1.57" effort="200" velocity="3"/>
    <dynamics damping="10" friction="1"/>
  </joint>
  <link name="forearm_link">
    <inertial>
      <origin xyz="-0.1739 -0.033049 -2.6025E-06" rpy="0 0 0"/>
      <mass value="2.275"/>
      <inertia ixx="0.002241" ixy="-0.00011285" ixz="4.2229E-07" iyy="0.011808" iyz="-2.6928E-07" izz="0.011655"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://ur5/forearm_link.stl"/>
      </geometry>
      <material name="White">
        <color rgba="1 0 1 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://ur5/forearm_link.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="elbow_joint" type="revolute">
    <origin xyz="-0.425 0 0" rpy="0 0 0"/>
    <parent link="upper_arm_link"/>
    <child link="forearm_link"/>
    <axis xyz="0 1 0"/>
    <limit lower="-6.2832" upper="6.2832" effort="200" velocity="3"/>
    <dynamics damping="10" friction="1"/>
  </joint>
  <link name="wrist_1_link">
    <inertial>
      <origin xyz="0.003753 0.044978 1.2618E-05" rpy="0 0 0"/>
      <mass value="1.219"/>
      <inertia ixx="0.00041141" ixy="4.2784E-05" ixz="-2.8598E-07" iyy="0.00036744" iyz="4.089E-07" izz="0.000387"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://ur5/wrist_1_link.stl"/>
      </geometry>
      <material name="Red">
        <color rgba="1 0 1 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://ur5/wrist_1_link.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="wrist_1_joint" type="revolute">
    <origin xyz="-0.39225 0 0" rpy="0 0 0"/>
    <parent link="forearm_link"/>
    <child link="wrist_1_link"/>
    <axis xyz="0 -1 0"/>
    <limit lower="-6.2832" upper="6.2832" effort="200" velocity="3"/>
    <dynamics damping="10" friction="1"/>
  </joint>
  <link name="wrist_2_link">
    <inertial>
      <origin xyz="-0.083749 -0.0037532 -1.2446E-05" rpy="0 0 0"/>
      <mass value="1.219"/>
      <inertia ixx="0.00036744" ixy="4.2782E-05" ixz="4.1013E-07" iyy="0.0004114" iyz="-2.8709E-07" izz="0.000387"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://ur5/wrist_2_link.stl"/>
      </geometry>
      <material name="Orange">
        <color rgba="1 0.5 0 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://ur5/wrist_2_link.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="wrist_2_joint" type="revolute">
    <origin xyz="0 0.056149 0" rpy="0 0 0"/>
    <parent link="wrist_1_link"/>
    <child link="wrist_2_link"/>
    <axis xyz="1 0 0"/>
    <limit lower="-6.2832" upper="6.2832" effort="200" velocity="3"/>
    <dynamics damping="10" friction="1"/>
  </joint>
  <link name="wrist_3_link">
    <inertial>
      <origin xyz="-7.8282E-05 0.018179 0.00011982" rpy="0 0 0"/>
      <mass value="0.1879"/>
      <inertia ixx="6.8269E-05" ixy="5.7754E-08" ixz="-3.2009E-07" iyy="0.00011604" iyz="-8.84E-08" izz="6.8551E-05"/>
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://ur5/wrist_3_link.stl"/>
      </geometry>
      <material name="White">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://ur5/wrist_3_link.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="wrist_3_joint" type="revolute">
    <origin xyz="-0.09492 0.041547 0" rpy="3.1416 0 3.1416"/>
    <parent link="wrist_2_link"/>
    <child link="wrist_3_link"/>
    <axis xyz="0 -1 0"/>
    <limit lower="-6.2832" upper="6.2832" effort="100" velocity="3"/>
    <dynamics damping="10" friction="1"/>
  </joint>
  <joint name="gripper_joint1" type="revolute">
    <axis xyz="1 0 0"/>
    <parent link="wrist_3_link"/>
    <child link="vacuum_gripper1"/>
    <origin rpy="1.570 0 0" xyz="0 0.04 0"/>
    <limit effort="50" velocity="50" lower="0" upper="0"/>
    <dynamics damping="0.0" friction="1000"/>
  </joint>
  <link name="vacuum_gripper1">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="0.0 0.0 0.0"/>
      </geometry>
    </collision>
    <gravity>0</gravity>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius="0.02" length="0.015"/>
      </geometry>
      <material name="Red">
        <color rgba="1 0 1 1"/>
      </material>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0.000000 0.000000 0.000000"/>
      <mass value="0.0001"/>
      <inertia ixx="1e-08" ixy="0" ixz="0" iyy="1e-08" iyz="0" izz="1e-08"/>
    </inertial>
  </link>
  <joint name="camera_joint" type="fixed">
    <axis xyz="0 1 0"/>
    <origin xyz="0.03 0.04 0" rpy="0 0 1.508"/>
    <parent link="wrist_3_link"/>
    <child link="camera_link"/>
  </joint><!-- Camera link -->
  <link name="camera_link">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
    </collision>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <material name="Black"/>
    </visual>
    <inertial>
      <mass value="1e-5"/>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6"/>
    </inertial>
  </link><!-- World and base links -->
  <link name="world"><!-- No inertial properties required --></link>
  <joint name="world_to_base" type="fixed">
    <parent link="world"/>
    <child link="base_link"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
</robot>